src/node/internal/internal_performance_entry.ts

```diff
@@ -0,0 +1,98 @@
+import { ERR_ILLEGAL_CONSTRUCTOR } from 'node-internal:internal_errors';
+import { validateThisInternalField } from 'node-internal:validators';
+import { inspect, type InspectOptions } from 'node-internal:internal_inspect';
+import { kEnumerableProperty } from 'node-internal:internal_utils';
+
+import type {
+  PerformanceEntry as _PerformanceEntry,
+  EntryType,
+} from 'node:perf_hooks';
+
+const kInspect = inspect.custom;
+
+const kName = Symbol('PerformanceEntry.Name');
+const kEntryType = Symbol('PerformanceEntry.EntryType');
+const kStartTime = Symbol('PerformanceEntry.StartTime');
+const kDuration = Symbol('PerformanceEntry.Duration');
+export const kSkipThrow = Symbol('kSkipThrow');
+
+export class PerformanceEntry implements _PerformanceEntry {
+  [kName]: string | undefined;
+  [kEntryType]: string | undefined;
+  [kStartTime]: number | undefined;
+  [kDuration]: number | undefined;
+
+  constructor(
+    skipThrowSymbol: symbol | undefined = undefined,
+    name: string | undefined = undefined,
+    type: EntryType | undefined = undefined,
+    start: number | undefined = undefined,
+    duration: number | undefined = undefined
+  ) {
+    if (skipThrowSymbol !== kSkipThrow) {
+      throw new ERR_ILLEGAL_CONSTRUCTOR();
+    }
+
+    this[kName] = name;
+    this[kEntryType] = type;
+    this[kStartTime] = start;
+    this[kDuration] = duration;
+  }
+
+  get name(): string {
+    validateThisInternalField(this, kName, 'PerformanceEntry');
+    return this[kName] as string;
```

Comment:
If kName is undefined, we're just casting it to a string. Feels a bit weird so I just want to confirm this is what node does.